src/gray_scott_model.rs

```diff
@@ -41,118 +34,117 @@ impl ReactionDiffusionSystem {
         assert!(
             (0..self.len()).contains(&index),
             "index {} is not in range 0..{}!",
             index,
-            self.u.len()
+            self.len()
         );
 
-        let cs_vec = match cs {
-            ChemicalSpecies::U => &self.u,
-            ChemicalSpecies::V => &self.v,
-        };
-
-        *cs_vec
-            .get(index)
-            .unwrap_or_else(|| panic!("Tried to get {:?}[{}] but failed.", cs, index))
+        *self.uvs.get(index).expect("couldn't get uv by index")
     }
 
     pub fn len(&self) -> usize {
         (self.height * self.width) as usize
     }
 
-    pub fn set(&mut self, cs: ChemicalSpecies, x: isize, y: isize, v: f32) {
-        let index = get_wrapping_index(x, y, self.width, self.height);
-        let cs_vec = match cs {
-            ChemicalSpecies::U => &mut self.u,
-            ChemicalSpecies::V => &mut self.v,
-        };
-
-        let index = index % cs_vec.len();
+    pub fn set(&mut self, x: isize, y: isize, v: (f32, f32)) {
+        let index = get_wrapping_index(x, y, self.width as usize, self.height as usize);
+        let index = index % self.len();
+        let v = (v.0.clamp(-1.0, 1.0), v.1.clamp(-1.0, 1.0));
 
-        cs_vec[index] = v.clamp(-1.0, 1.0)
+        self.uvs[index] = v;
     }
 
-    pub fn get_laplacian(&self, cs: ChemicalSpecies, x: isize, y: isize) -> f32 {
-        0.05 * self.get(cs, x - 1, y - 1)
-            + 0.2 * self.get(cs, x - 1, y)
-            + 0.05 * self.get(cs, x - 1, y + 1)
-            + 0.2 * self.get(cs, x, y - 1)
-            + -1.0 * self.get(cs, x, y)
-            + 0.2 * self.get(cs, x, y + 1)
-            + 0.05 * self.get(cs, x + 1, y - 1)
-            + 0.2 * self.get(cs, x + 1, y)
-            + 0.05 * self.get(cs, x + 1, y + 1)
+    pub fn get_laplacian(&self, x: isize, y: isize) -> (f32, f32) {
+        (
+            (0.05 * self.get(x - 1, y - 1).0
+                + 0.2 * self.get(x - 1, y).0
+                + 0.05 * self.get(x - 1, y + 1).0
+                + 0.2 * self.get(x, y - 1).0
+                + -1.0 * self.get(x, y).0
+                + 0.2 * self.get(x, y + 1).0
+                + 0.05 * self.get(x + 1, y - 1).0
+                + 0.2 * self.get(x + 1, y).0
+                + 0.05 * self.get(x + 1, y + 1).0),
+            (0.05 * self.get(x - 1, y - 1).1
+                + 0.2 * self.get(x - 1, y).1
+                + 0.05 * self.get(x - 1, y + 1).1
+                + 0.2 * self.get(x, y - 1).1
+                + -1.0 * self.get(x, y).1
+                + 0.2 * self.get(x, y + 1).1
+                + 0.05 * self.get(x + 1, y - 1).1
+                + 0.2 * self.get(x + 1, y).1
+                + 0.05 * self.get(x + 1, y + 1).1),
+        )
     }
 
-    pub fn update(&mut self, delta_t: f32) {
-        // delta_t is in seconds, so we convert to milliseconds (which the diffusion calculation expects)
-        let delta_t = delta_t * 100.0;
-
+    pub fn update(&mut self, mut _delta_t: f32) {
```

Comment:
with changes to to this function does it always just return 1ms now?

Reply:
yeah, this is a whole thing that I decided to just sidestep for now. The reaction diffusion algorithm can accept timesteps up to 1.0, but if it goes higher than that, you have to run the update multiple times per tick in order to make it go faster. I wrote a loop thing to allow for that but it ends up taking too much CPU and I'm not sure how to tackle the issue. I'm just ignoring it for now by always passing 1.0 so that it internally runs at max speed always and is only really limited by fps.